interfaces/ldconfig/spec.go

```diff
@@ -87,7 +87,7 @@ func (spec *Specification) AddConnectedPlug(iface interfaces.Interface, plug *in
 
 // AddConnectedSlot records ldconfig-specific side-effects of having a connected slot.
 func (spec *Specification) AddConnectedSlot(iface interfaces.Interface, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
-	if !interfaces.IsTheSystemSnap(plug.Snap().InstanceName()) {
+	if !interfaces.IsTheSystemSnap(slot.Snap().InstanceName()) {
```

Comment:
this change seems definitely wrong we probably need to do something else and move the check only if the interface has the definer no?

Reply:
Also in either case this *definitely* warrants a comment

Reply:
Right, the check really needs to happen only for interfaces supporting ldconfig, I've changed this now and reverted to check for the plug here too.